src/components/AgreementSections/FormInfo.js

```diff
@@ -24,35 +24,36 @@ export default class FormInfo extends React.Component {
   };
 
   state = {
-    dataOptions: {
-      agreementStatusValues: [],
-      renewalPriorityValues: [],
-      isPerpetualValues: [],
-    }
+    agreementStatusValues: [],
+    isPerpetualValues: [],
+    renewalPriorityValues: [],
   }
 
   // Prepend an empty value to each set of dropdown options to facilitate
   // unselecting a value.
   static getDerivedStateFromProps(props, state) {
     const { data } = props;
-    const dataOptions = {};
+    const newState = {};
+
+    if (data.agreementStatusValues.length !== state.agreementStatusValues) {
```

Comment:
Sorry, these all should be `if (... != ....length)`. We're comparing whether the length of the array we just got from `props` has the same length (and thus is the same array) as the array we've stored in `state`.